File: equipes/maroua_team/P2.py
from pathlib import Path


def max_houses(N: int, B: int, prices: list[int]):
    """
    Calcule le nombre maximal de maisons que l'on peut acheter avec un budget
    donné.

    Trie les prix par ordre croissant, puis achète autant de maisons que
    possible en commençant par les moins chères jusqu'à ce que le budget
    soit dépassé.

    Args:
        N (int): Nombre de maisons disponibles.
        B (int): Budget total disponible.
        prices (list[int]): Liste des prix des maisons.

    Returns:
        int: Nombre maximal de maisons que l'on peut acheter avec le budget
        donné.
    """
    prices.sort()  # Triez les prix par ordre croissant
    houses_to_buy = 0
    total_price = 0

    for price in prices:
        if total_price + price <= B:
            houses_to_buy += 1
            total_price += price
        else:
            break

    return houses_to_buy


def main():
    """
    Fonction principale pour exécuter le programme.

    Lecture des données d'entrée à partir du fichier input.txt,
    exécution de la fonction max_houses pour chaque jeu de données,
    puis enregistrement des résultats dans le fichier output.txt.
    """
    with open(
        Path(__file__).resolve().parent.parent.parent / "P2/input.txt", "r"
    ) as file:
        tests = file.readlines()
    root = Path(__file__).resolve().parent.parent.parent
    output_file = root / "P2/output.txt"
    with open(output_file, "w") as output:
        for i in range(1, len(tests), 2):  # Parcourez les tests par pas de 2
            N, B = map(int, tests[i].strip().split())
            prices = list(map(int, tests[i + 1].strip().split()))
            result = max_houses(N, B, prices)
            output.write(f"Test {i//2 + 1}: {result}\n")

    print("Résultats enregistrés dans le fichier output.txt.")


if __name__ == "__main__":
    main()

File: equipes/maroua_team/P2.r
max_houses <- function(N, B, prices) {
  prices <- sort(prices)  # Triez les prix par ordre croissant
  houses_to_buy <- 0
  total_price <- 0

  for (price in prices) {
    if (total_price + price <= B) {
      houses_to_buy <- houses_to_buy + 1
      total_price <- total_price + price
    } else {
      break
    }
  }

  return(houses_to_buy)
}

main <- function() {
  tests <- readLines("data/P2/input.txt")

  output_file <- "data/P2/output.txt"
  sink(output_file)  # Rediriger la sortie vers le fichier output.txt

  for (i in seq(1, length(tests), by=2)) {
    line <- strsplit(tests[i], " ")[[1]]
    N <- as.integer(line[1])
    B <- as.integer(line[2])

    prices <- as.integer(strsplit(tests[i + 1], " ")[[1]])
    
    result <- max_houses(N, B, prices)
    cat("Test", (i + 1) / 2, ":", result, "\n")
  }

  sink()  # Arrêter la redirection de la sortie vers le fichier
  cat("Résultats enregistrés dans le fichier output.txt.\n")
}

main()
